ci/fix_paths.py

```diff
@@ -17,15 +18,21 @@ def main():
 
     # HDF5_DIR is not set when we're testing wheels; these should already have
     # the necessary libraries bundled in.
-    if platform.startswith('win') and hdf5_path is not None:
+    if sys_platform.startswith('win') and hdf5_path is not None:
         for f in glob(pjoin(hdf5_path, 'lib/*.dll')):
             copy(f, pjoin(sitepackagesdir, 'h5py', basename(f)))
             print("Copied", f)
 
         zlib_root = os.environ.get("ZLIB_ROOT")
         if zlib_root:
-            f = pjoin(zlib_root, 'bin_release', 'zlib.dll')
-            copy(f, pjoin(sitepackagesdir, 'h5py', 'zlib.dll'))
+            arch = platform.machine().lower()
+            if arch in ("arm64", "aarch64"):
+                f = pjoin(zlib_root, 'bin', 'zlib1.dll')
+            elif arch in ("amd64", "x86_64"):
+                f = pjoin(zlib_root, 'bin_release', 'zlib.dll')
+            else:
+                raise RuntimeError(f"Unexpected architecture detected: {arch}")
```

Comment:
So the error message clearly shows the unaltered value and where it comes from
```suggestion
                raise RuntimeError(f"Unexpected architecture detected: {platform.machine()=}")
```